### PR TITLE
Improve TypeScript declarations file

### LINKED ISSUE #581: error TS2497: Module ''jimp'' resolves to a non-module entity and cannot be imported using this construct.
Hi,

I cannot import jimp with typescript, getting error. Details are below.

# Expected Behavior
There shouldn't be a compilation-time error.

# Current Behavior

There is a compilation-time error.
`error TS2497: Module ''jimp'' resolves to a non-module entity and cannot be imported using this construct. import * as jimp from 'jimp';`

Also,  `import jimp from 'jimp';` doesn't work because it imports the jimp into jimp_1 and becomes not usable...

# Failure Information (for bugs)

## Steps to Reproduce
Import jimp in a typescript file with `import * as jimp from 'jimp';`

- Jimp Version: ^0.3.9
- Operating System: MacOS
- Node version: v8.11.3
- TypeScript version: 3.0.1

## Failure Logs

`error TS2497: Module ''jimp'' resolves to a non-module entity and cannot be imported using this construct. import * as jimp from 'jimp';`

### PR DESCRIPTION
**Pull request by Favna**
# What's Changing and Why

Addresses but does not fully close #581

Proper way to import is now `import * as X from 'jimp'` wherein X can be
whatever the end-user wants, in most cases `Jimp`.

As stated on the aforementioned issue another PR might come with a more thorough fix not requiring `// @ts-ignore`

## What else might be affected

Method of importing with TypeScript, possible requires doc updates.

## Tasks

- [ ] Add tests
- [x] Update Documentation
- [ ] Update `jimp.d.ts`
- [x] Add [SemVer](https://semver.org/) Label

### CODE REVIEW

File: packages/jimp/jimp.d.ts [hunk 1]
```diff
@@ -561,6 +561,6 @@ declare namespace Jimp {
   }
 }
 
-declare module 'jimp' {
-  export = Jimp.Jimp;
-}
+// @ts-ignore
```
Comment (DanielRosenwasser) [inline packages/jimp/jimp.d.ts]: No.

**Comment by Favna** [comment]
@DanielRosenwasser @hipstersmoothie I've reworked it so it no longer requires the `ts-ignore` to satisfy the IDEs. Would appreciate another review. Copying from my Git commit message:

I have losely based this on knowledge gained from analyzing other declarations files, primarily `@types/better-sqlite3`. Using this formatting for declaration file allows for `import * as Jimp from 'jimp'` importing which is a proper way when using TypeScript as it primarily recommends ES6 imports. It will require setting the `allowSyntheticDefaultImports` compiler option to `true`, however this is a very small issue as many packages on npm (including aforementioned better-sqlite3) require this.

- Like previous commit this ensures we no longer need to use `import
Jimp = require('jimp'). As has been agreed upon, using `require` when
working with ES6 modules is just wrong in code style.

- Unlike previous commit no more ts-ignore, yay!

- It is maintainable. New types or interface come at the bottom and any
new or modifications to functions and constants go in the topmost
interface. We'll never have to touch the declared const or the export.

---

Lastly, I would really appreciate a new version to be pushed to npm if you guys decide to merge this. Due to your building setup I can't just `yarn add favna/jimp#fix/typedeclarations` to install my own branch so until a new proper build is released I'll have to use the old method of importing which also blocks me from updating my documentation as the jsdoc2md compiler cannot handle the `= require` structure.

**Comment by Favna** [comment]
Oh and one final note, if you guys decide to merge this I think #581 can be closed since the importing and exporting should be entirely fixed but I'll let that upon your judgement.

**Comment by hipstersmoothie** [comment]
@Favna I will release once @DanielRosenwasser gives a 👍

**Comment by DanielRosenwasser** [comment]
Okay, I think that this change is appropriate. Users can use

```js
import jimp = require("jimp")
```

to import the module.

Alternatively, if they have `esModuleInterop` they can use a default import:

```js
import jimp from "jimp";
```

**Comment by DanielRosenwasser** [comment]
For the record, because Jimp is callable, `import * as jimp from "jimp"` shouldn't be used.

**Comment by hipstersmoothie** [comment]
@Favna can you add this to the docs in the "jimp" package readme? After that I'll get it merged and released.

**Comment by Favna** [comment]
Ow.. I see you merged already. Sorry I didn't add the usage doc to the readme yet. Long work days means I only just got home and you guys commented at my 2AM :\

I'll add the usage docs right now and create a new PR I guess.

**Pull request merged**